src/test/java/com/netflix/playback/features/impl/PlaybackDiagnosticsImplTest.java

```diff
@@ -0,0 +1,104 @@
+package com.netflix.playback.features.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.playback.features.model.PeriodCompletedCallback;
+import com.netflix.playback.features.model.PlaybackRequest;
+
+/**
+ * Tests {@link PlaybackDiagnosticsImpl}.
+ */
+public class PlaybackDiagnosticsImplTest {
+
+    private static Logger logger = LoggerFactory.getLogger(TimerImplTest.class);
+
+    private int periodLength = 100;
```

Comment:
Currently the PlaybackDiagnosticsImplTest is written more of a end to end test or kind of integration test.
A more unit test way is mock the Timer object and manually trigger callback as needed to make it go to the next period. This way there is no need for waiting on latch or for 100 milliseconds.